governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals/Proposal.tsx

```diff
@@ -140,25 +146,30 @@ export const Proposal = ({
   const [isTransactionLoading, setIsTransactionLoading] = useState(false)
   const { cluster: contextCluster } = useContext(ClusterContext)
   const multisigCluster = getMultisigCluster(contextCluster)
-  const targetClusters: (PythCluster | 'unknown')[] = []
+  const targetClusters: (SvmCluster | 'unknown')[] = []
   instructions.map((ix) => {
     if (!(ix instanceof WormholeMultisigInstruction)) {
       targetClusters.push(multisigCluster)
     } else if (
       ix instanceof WormholeMultisigInstruction &&
       ix.governanceAction instanceof ExecutePostedVaa
     ) {
-      ix.governanceAction.instructions.map((ix) => {
-        const remoteClusters: PythCluster[] = [
+      ix.governanceAction.instructions.map((innerIx) => {
+        const remoteClusters: SvmCluster[] = [
           'pythnet',
           'pythtest-conformance',
           'pythtest-crosschain',
+          'eclipse_mainnet',
         ]
         for (const remoteCluster of remoteClusters) {
           if (
             multisigCluster === getMultisigCluster(remoteCluster) &&
-            (ix.programId.equals(getPythProgramKeyForCluster(remoteCluster)) ||
-              ix.programId.equals(SystemProgram.programId))
+            ix.governanceAction?.targetChainId === remoteCluster &&
+            (innerIx.programId.equals(
+              getPythProgramKeyForCluster(getPythnetCluster(remoteCluster))
+            ) ||
+              innerIx.programId.equals(SystemProgram.programId) ||
+              isSolanaPullProgram(innerIx.programId))
```

Comment:
The changes in this file, just feel just super hacky. I'm not sure they're worth it.
The problem here is that this feature of target network was designed to distinguish pythnet, pythtest-conformance and pythtest-crosschain in price feeds council. This is important because they all have the same chain id (26).

I think making it work for every network would be big refactor and anyway for all the other networks they each have their own chain id so you will be able to distinguish them easily at the instruction level.

Reply:
If you think it's weird to show target network unknown or no target network detected, just don't show any text if the cluster doesn't get resolved.